modules/cart/src/CartManager.php

```diff
@@ -89,6 +109,19 @@ public function emptyCart(OrderInterface $cart, $save_cart = TRUE) {
   /**
    * {@inheritdoc}
    */
+  public function finalizeCart(OrderInterface $cart, $save_cart = TRUE) {
+    $cart->cart = FALSE;
+    $this->cartSession->deleteCartId($cart->id());
```

Comment:
This prevents anonymous checkout from working. Maybe we move the session removal out of this method, and just keep it in the checkout form submit. What do you think @bojanz 

<img width="907" alt="access_denied___drupal_and_google_hangouts_-_matt_commerceguys_com" src="https://cloud.githubusercontent.com/assets/3698644/18682781/2dec8c78-7f33-11e6-9d5f-4f3efa877b8d.png">